### DISCUSSION TITLE
Non-Admin backup/restore operations performance

### DISCUSSION THREAD

**Discussion by shubham-pampattiwar** [migtools/oadp-non-admin#30] [General]
Due to the current limitation that a single instance of velero processes one single backup/restore request at a time, the non-admin backup/restore ops would not be performant.
As an alternative multiple instance of OADP with NAC deployment can be considered to address the performance concerns (Given that each NAC instance will be tied to one OADP Operator instance i.e OADP:NAC :: 1:1)

**Comment by shubham-pampattiwar** [comment]
Related Discussion thread: https://github.com/migtools/oadp-non-admin/discussions/31

**Comment by kaovilai** [comment]
Hopefully upgrades can be done across multiple veleroes around the same time with no crd conflict. Maybe guidance to pause all schedules and wait till all backups complete b4 upgrade.

**Comment by dankenigsberg** [comment]
I'm not familiar with the design of Velero or OADP, but from the sound of it, it seems that the Velero limitation is the one that should be fixed. Having each user install their OADP sounds like a handful. Regardless of the implementation, we must not let a namespace-specific backup DoS a cluster-wide one.
